### PR TITLE
Analytics: Prevent tracks calls when DO_NOT_TRACK is truthy

### PR DESCRIPTION
**Pull request by jblz**
## Description

Provide a mechanism to opt-out of telemetry / usage tracking. If the `DO_NOT_TRACK` environment variable is truthy, prevent the sending of data to the "tracks" system.

See: https://consoledonottrack.com/

PLAT-1054

## Steps to Test

1. Check out PR.
1. Run `npm run build`
1. Run the automated tests (`npm run test __tests__/lib/analytics/clients/tracks.js`)
    1. The added test logic should run & be logically-sound
    1. All tests should pass
1. Apply [this patch](https://gist.github.com/jblz/dd87fb4ca0f9f4cde43e0125bc998099) to manually test
1. Run `npm run build`
1. Run `DO_NOT_TRACK=1 ./dist/bin/vip-app-list.js`
    1. Verify your available apps are listed
    1. Verify all events are labeled `wouldNotTrack`
1. Run `./dist/bin/vip-app-list.js` (NOTE: this assumes you do not have the env var set system-wide.  If you do, override it like `DO_NOT_TRACK=0 ./dist/bin/vip-app-list.js` here.)
    1. Verify your available apps are listed
    1. Verify all events are labeled `wouldTrack`

### CODE REVIEW

File: src/lib/analytics/clients/tracks.js [hunk 1]
```diff
@@ -85,6 +85,10 @@ export default class Tracks implements AnalyticsClient {
 	}
 
 	send( extraParams: {} ): Promise<Response> {
+		if ( process.env.DO_NOT_TRACK ) {
+			return Promise.resolve( 'tracks disabled per DO_NOT_TRACK variable' );
```
Comment (mjangda) [inline src/lib/analytics/clients/tracks.js:91]: We're no longer returning the correct type as indicated by the flow type for the function (`Promise<Response>`). Should we update that?

**Comment by mjangda** [comment]
Alternate way to verify: `DO_NOT_TRACK=1 DEBUG=* ./dist/bin/vip-app-list.js`

`debug` will no longer output `@automattic/vip:analytics:clients:tracks send()` entries.

File: src/lib/analytics/clients/tracks.js (hunk 1 above)
Comment (mjangda) [inline src/lib/analytics/clients/tracks.js:91]: Let's add a `debug` call for logging/debugging purposes.

```suggestion
			debug( 'send() => skipping per DO_NOT_TRACK variable' );

			return Promise.resolve( 'tracks disabled per DO_NOT_TRACK variable' );
```

**Review by mjangda — COMMENTED** [review]
Spotted two minor things but otherwise this looks great!

**Review by mjangda — DISMISSED** [review]
:ship:

**Comment by elazzabi** [comment]
I think we should add this somewhere to the README. Any ideas on where we can add it? Maybe a new heading or something 🤔

**Comment by jblz** [comment]
Good suggestion, @elazzabi.  Added a README section in 
233e4dc

**Pull request merged**